File: Exhibition.Core/Models/Directive.cs


namespace Exhibition.Core.Models
{
    using Newtonsoft.Json;
    public class Directive
    {

        /// <summary>
        /// 
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public IBaseTerminal Terminal { get; set; }
        /// <summary>
        /// 
        /// </summary>
        [JsonProperty("DefaultWindow",NullValueHandling = NullValueHandling.Ignore)]
        public Window DefaultWindow { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public Resource[] Resources { get; set; }
    }

    public class DirectiveforTransmit
    {
        /// <summary>
        /// 
        /// </summary>
        public string Name { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public string Description { get; set; }

        /// <summary>
        /// 
        /// </summary>
        public string Terminal { get; set; }        
        
        public Window DefaultWindow { get; set; }
        /// <summary>
        /// 
        /// </summary>
        public Resource[] Resources { get; set; }
    }

}

File: Exhibition.Portal.Web/src/app/models/Directive.ts
import { Terminal,Window } from "./terminal";
import { Resource } from "./Resource";

export class Directive{
    public type:number;
    public terminal:Terminal;
    public window:Window;    
    public name:string;
    public resource:Resource;
    public description:string;
}

export class OptionModel{
    public key:string;
    public text:string;
}

export class DirectiveforEditing{
    public type:number;
    public terminalIp:string;
    public windowId:number;    
    public name:string;
    public resourceFullName:string;
    public description:string;
}